src/tty.c

```diff
@@ -40,7 +40,13 @@ METAL_CONSTRUCTOR(metal_tty_init) {
  * provide a shim that eats all the characters so we can ensure that everything
  * can link to metal_tty_putc. */
 int nop_putc(int c) __attribute__((section(".text.metal.nop.putc")));
-int nop_putc(int c) { return -1; }
+// Use a customizable NOP hint instruction so that a post-processor parser can
+// look for this instruction, and use the value in a0 as the character to be
```

Comment:
For documentation purposes, it might be worth clarifying that "the value in `a0`" will have undergone some extensions (`char` -> `int` -> `int{32,64}_t`) and only the least significant 8 bits are relevant.

Reply:
Promotion from unsigned char to 32/64-bit int will leave the value unchanged, though, because it's an unsigned char.

Do you want me to explicitly AND it with 0xff before doing the slli, to make it completely clear?

Reply:
Here's code that works for the current output from Verilator +verbose logging:

awk '
/inst=.01151013/ {
  char=$(NF-4);
  char = "0x"substr(char,21, 2);
  char=strtonum(char);
  printf "%c", char;
}' $*

Or the one-liner version, for easier cut-and-pasting:

awk '/inst=.01151013/{printf "%c",strtonum("0x"substr($(NF-4),21,2));}' <filename>

Reply:
Agreed: unsigned char will get zero extended to 32 bits, so a (possible) subsequent sign extension to 64 bits is just another zero extension. Feel free to disregard my comment.